src/spec.rs

```diff
@@ -145,52 +351,83 @@ pub struct Zip64CentralDirectoryEnd {
 }
 
 impl Zip64CentralDirectoryEnd {
+    pub fn parse<T: Read>(reader: &mut T) -> ZipResult<Zip64CentralDirectoryEnd> {
+        let Zip64CDEBlock {
+            // record_size,
+            version_made_by,
+            version_needed_to_extract,
+            disk_number,
+            disk_with_central_directory,
+            number_of_files_on_this_disk,
+            number_of_files,
+            central_directory_size,
+            central_directory_offset,
+            ..
+        } = Zip64CDEBlock::parse(reader)?;
+        Ok(Self {
+            version_made_by,
+            version_needed_to_extract,
+            disk_number,
+            disk_with_central_directory,
+            number_of_files_on_this_disk,
+            number_of_files,
+            central_directory_size,
+            central_directory_offset,
+        })
+    }
+
     pub fn find_and_parse<T: Read + Seek>(
         reader: &mut T,
         nominal_offset: u64,
         search_upper_bound: u64,
     ) -> ZipResult<Vec<(Zip64CentralDirectoryEnd, u64)>> {
         let mut results = Vec::new();
-        let mut pos = search_upper_bound;
-
-        while pos >= nominal_offset {
-            reader.seek(io::SeekFrom::Start(pos))?;
-
-            if reader.read_u32_le()? == ZIP64_CENTRAL_DIRECTORY_END_SIGNATURE {
-                let archive_offset = pos - nominal_offset;
-
-                let _record_size = reader.read_u64_le()?;
-                // We would use this value if we did anything with the "zip64 extensible data sector".
-
-                let version_made_by = reader.read_u16_le()?;
-                let version_needed_to_extract = reader.read_u16_le()?;
-                let disk_number = reader.read_u32_le()?;
-                let disk_with_central_directory = reader.read_u32_le()?;
-                let number_of_files_on_this_disk = reader.read_u64_le()?;
-                let number_of_files = reader.read_u64_le()?;
-                let central_directory_size = reader.read_u64_le()?;
-                let central_directory_offset = reader.read_u64_le()?;
-
-                results.push((
-                    Zip64CentralDirectoryEnd {
-                        version_made_by,
-                        version_needed_to_extract,
-                        disk_number,
-                        disk_with_central_directory,
-                        number_of_files_on_this_disk,
-                        number_of_files,
-                        central_directory_size,
-                        central_directory_offset,
-                    },
-                    archive_offset,
-                ));
-            }
-            if pos > 0 {
-                pos -= 1;
-            } else {
+
+        const END_WINDOW_SIZE: usize = 2048;
```

Comment:
I'm pretty sure these windows need to overlap by at least `size_of_val(&ZIP64_CENTRAL_DIRECTORY_END_SIGNATURE) - 1` bytes, in case the signature straddles a 2KiB boundary.

Reply:
Wow, great catch! Thanks! I will fix this.

Reply:
Going to try to add a test case for this if it isn't too hard.

Reply:
Done! See e.g. https://github.com/zip-rs/zip2/pull/93/commits/5a08eaee4d0498383e8ce387ad1d3448a79f7ff2